numpydoc/validate.py

```diff
@@ -506,9 +578,16 @@ def validate(obj_name, validator_cls=None, **validator_kwargs):
     else:
         doc = validator_cls(obj_name=obj_name, **validator_kwargs)
 
+    # lineno is only 0 if we have a module docstring in the file and we are
+    # validating that, so we change to 1 for readability of the output
+    ignore_validation_comments = extract_ignore_validation_comments(
+        doc.source_file_name
+    ).get(doc.source_file_def_line or 1, [])
```

Comment:
... I think this needs a `if doc.source_file_name is not None` perhaps?

Reply:
Thanks for the heads-up!

Reply:
How did the Sphinx validation behave before when the `source_file_name` is `None`? Do you still run the validation?

Reply:
Yes. We're generate some doc strings on the fly, I assume that's what's breaking this but not sure